### PR TITLE
Feature v0.14.6 daemon restart

### PR DESCRIPTION
**Pull request by mattkasun**
replace calls to systemctl restart netclient (and other variants) with SIGHUP signal

### CODE REVIEW

**Comment by mattkasun** [comment]
Testing:
run client on all os
execute functions that generate client restart (pull, join, leave,  UI node changes)
  look for recieved reset/restarting daemon in log file with no errors

File: netclient/ncutils/pid.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package ncutils
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+)
+
+// PIDFILE - path/name of pid file
+const PIDFILE = "/var/run/netclient.pid"
```
Comment (0xdcarns) [inline netclient/ncutils/pid.go:10]: will this account for a windows path..?

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: netclient/ncutils/pid.go (hunk 1 above)
Comment (mattkasun) [reply netclient/ncutils/pid.go:10]: added check for windows

**Comment by theguy951357** [comment]
I have clients going into error. 
If I do  netclient pull without specifying a network, I get an error in the journalctl: 
Jul 13 01:44:44 egressQA systemd[1]: netclient.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Jul 13 01:44:44 egressQA systemd[1]: netclient.service: Failed with result 'exit-code'.
Jul 13 01:45:55 fedoraQA netclient[590006]: [netclient] 2022-07-13 01:45:55 checking for netclient updates...
Jul 13 01:45:55 fedoraQA netclient[590006]: [netclient] 2022-07-13 01:45:55 schema of network daemon-lite is out of date and cannot be updated
if I do  netclient pull and specify the network, the daemon shuts down and I have to manually start it back up.

**Comment by mattkasun** [comment]
reverted to handling restarts the old way for windows.

still an issue with negative waitgroups when doing pull without specifying network

**Comment by mattkasun** [comment]
gui build failure due to ??out of date apt mirror at azure?? for runner.

**Review by afeiszli — APPROVED** [review]

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**